undefined reference to `__log_current_const_data_get' when using LOG_MODULE_DECLARE

With this change: https://github.com/nashif/zephyr/commit/e94e0c987be5032bb6b76e32e6b45a4bcc43b702

I get
```
kernel/libkernel.a(userspace.c.obj): In function `_impl_k_object_alloc':
/home/nashif/Work/github/zephyr/kernel/userspace.c:228: undefined reference to `__log_current_const_data_get'
collect2: error: ld returned 1 exit status
```

--- a/include/logging/log.h
+++ b/include/logging/log.h
@@ -315,8 +315,13 @@ int log_printk(const char *fmt, va_list ap);
 		()/*Empty*/						\
 	)
 
-#define __DYNAMIC_MODULE_DECLARE(_name)				\
-	extern struct log_source_dynamic_data LOG_ITEM_DYNAMIC_DATA(_name)
+#define __DYNAMIC_MODULE_DECLARE(_name)					   \
+	extern struct log_source_dynamic_data LOG_ITEM_DYNAMIC_DATA(_name);\
+	static inline struct log_source_dynamic_data *			   \
+				__log_current_dynamic_data_get(void)	   \
+	{								   \
+		return &LOG_ITEM_DYNAMIC_DATA(_name);			   \
+	}
 
 #define _LOG_RUNTIME_MODULE_DECLARE(_name)			\
 	_LOG_EVAL(						\
@@ -325,9 +330,14 @@ int log_printk(const char *fmt, va_list ap);
 		()						\
 		)
 
-#define _LOG_MODULE_DECLARE(_name, _level)				\
+#define _LOG_MODULE_DECLARE(_name, _level)				     \
 	extern const struct log_source_const_data LOG_ITEM_CONST_DATA(_name) \
-	_LOG_RUNTIME_MODULE_DECLARE(_name)
+	_LOG_RUNTIME_MODULE_DECLARE(_name);				     \
+	static inline const struct log_source_const_data *		     \
+				__log_current_const_data_get(void)	     \
+	{								     \
+		return &LOG_ITEM_CONST_DATA(_name);			     \
+	}
 
 /**
  * @brief Macro for declaring a log module (not registering it).

--- a/tests/subsys/logging/log_core/src/log_core_test.c
+++ b/tests/subsys/logging/log_core/src/log_core_test.c
@@ -27,6 +27,8 @@ struct backend_cb {
 	size_t counter;
 	bool panic;
 	bool keep_msgs;
+	bool check_id;
+	u32_t exp_id[100];
 	bool check_timestamp;
 	u32_t exp_timestamps[100];
 	bool check_args;
@@ -39,6 +41,13 @@ static void put(struct log_backend const *const backend,
 	log_msg_get(msg);
 	struct backend_cb *cb = (struct backend_cb *)backend->cb->ctx;
 
+	if (cb->check_id) {
+		u32_t exp_id = cb->exp_id[cb->counter];
+
+		zassert_equal(log_msg_source_id_get(msg),
+			      exp_id,
+			      "Unexpected source_id");
+	}
 	if (cb->check_timestamp) {
 		u32_t exp_timestamp = cb->exp_timestamps[cb->counter];
 
@@ -293,6 +302,28 @@ static void test_log_panic(void)
 		      "Unexpected amount of messages received by the backend.");
 }
 
+/* extern function comes from the file which is part of test module. It is
+ * expected that logs coming from it will have same source_id as current
+ * module (this file).
+ */
+extern void test_func(void);
+static void test_log_from_declared_module(void)
+{
+	log_setup(false);
+
+	/* Setup log backend to validate source_id of the message. */
+	backend1_cb.check_id = true;
+	backend1_cb.exp_id[0] = LOG_CURRENT_MODULE_ID();
+
+	test_func();
+
+	while (log_process(false)) {
+	}
+
+	zassert_equal(1, backend1_cb.counter,
+		      "Unexpected amount of messages received by the backend.");
+}
+
 /*test case main entry*/
 void test_main(void)
 {
@@ -300,6 +331,7 @@ void test_main(void)
 			 ztest_unit_test(test_log_backend_runtime_filtering),
 			 ztest_unit_test(test_log_overflow),
 			 ztest_unit_test(test_log_arguments),
-			 ztest_unit_test(test_log_panic));
+			 ztest_unit_test(test_log_panic),
+			 ztest_unit_test(test_log_from_declared_module));
 	ztest_run_test_suite(test_log_list);
 }

--- a/tests/subsys/logging/log_core/src/test_module.c
+++ b/tests/subsys/logging/log_core/src/test_module.c
@@ -1,0 +1,19 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Test module for log core test
+ *
+ */
+
+#include <logging/log.h>
+LOG_MODULE_DECLARE(test);
+
+void test_func(void)
+{
+	LOG_ERR("test");
+}

--- a/tests/subsys/logging/log_core/testcase.yaml
+++ b/tests/subsys/logging/log_core/testcase.yaml
@@ -1,3 +1,5 @@
 tests:
   logging.log_core:
     tags: log_core logging
+    platform_exclude: altera_max10 qemu_nios2 nucleo_l053r8
+      nucleo_f030r8 quark_d2000_crb stm32f0_disco

